Src/BootCamp.Chapter/PcFactory.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using BootCamp.Chapter.Computer;
+
+namespace BootCamp.Chapter
+{
+    public abstract class PcFactory
+    {
+        protected Body body;
+        protected Cpu cpu;
+        protected Gpu gpu;
+        protected HardDisk disk;
+        protected Motherboard board;
+        protected Ram ram;
+
+        public DesktopComputer Assemble()
+        {
+            AssembleBody();
+            AssembleCpu();
+            AssembleGpu();
+            AssembleHardDisk();
+            AssembleMotherboard();
+            AssembleRam();
+            return new DesktopComputer(body, cpu, gpu, disk, board, ram);
+        }
+
+        public abstract void AssembleBody();
+        public abstract void AssembleCpu();
+        public abstract void AssembleGpu();
+        public abstract void AssembleHardDisk();
+        public abstract void AssembleMotherboard();
+        public abstract void AssembleRam();
```

Comment:
Naming nitpick - what exactly are these methods assembling? Maybe `PrepareXXX` might be better? 

Also, make these `protected`. As a rule, don't make something `public` until you've got a proven use case where it's necessary.